test/customer/customer.go

```diff
@@ -0,0 +1,316 @@
+
+	// Create a createdCustomer
+	createdCustomer, err := service.CreateCustomer(ctx, customer.CreateCustomerInput{
+		Namespace: s.namespace,
+		Customer: customer.Customer{
+			Name:           TestName,
+			PrimaryEmail:   &TestPrimaryEmail,
+			Currency:       &TestCurrency,
+			BillingAddress: &TestAddress,
+			UsageAttribution: customer.CustomerUsageAttribution{
+				SubjectKeys: TestSubjectKeys,
+			},
+			TaxProvider:       &TestTaxProvider,
+			InvoicingProvider: &TestInvoicingProvider,
+			PaymentProvider:   &TestPaymentProvider,
+		},
+	})
+
+	require.NoError(t, err, "Creating customer must not return error")
+
+	require.NotNil(t, createdCustomer, "Customer must not be nil")
+	require.Equal(t, s.namespace, createdCustomer.Namespace, "Customer namespace must match")
+	require.NotNil(t, createdCustomer.ID, "Customer ID must not be nil")
+	require.Equal(t, TestName, createdCustomer.Name, "Customer name must match")
+	require.Equal(t, &TestPrimaryEmail, createdCustomer.PrimaryEmail, "Customer primary email must match")
+	require.Equal(t, &TestCurrency, createdCustomer.Currency, "Customer currency must match")
+	require.Equal(t, &TestAddressCountry, createdCustomer.BillingAddress.Country, "Customer billing address country must match")
+	require.Equal(t, &TestAddressCity, createdCustomer.BillingAddress.City, "Customer billing address city must match")
+	require.Equal(t, &TestAddressLine1, createdCustomer.BillingAddress.Line1, "Customer billing address line1 must match")
+	require.Equal(t, &TestAddressLine2, createdCustomer.BillingAddress.Line2, "Customer billing address line2 must match")
+	require.Equal(t, &TestAddressPostalCode, createdCustomer.BillingAddress.PostalCode, "Customer billing address postal code must match")
+	require.Equal(t, &TestAddressPhoneNumber, createdCustomer.BillingAddress.PhoneNumber, "Customer billing address phone number must match")
+	require.Equal(t, TestSubjectKeys, createdCustomer.UsageAttribution.SubjectKeys, "Customer usage attribution subject keys must match")
+	require.Equal(t, &TestTaxProvider, createdCustomer.TaxProvider, "Customer tax provider must match")
+	require.Equal(t, &TestInvoicingProvider, createdCustomer.InvoicingProvider, "Customer invoicing provider must match")
+	require.Equal(t, &TestPaymentProvider, createdCustomer.PaymentProvider, "Customer payment provider must match")
+
+	// Test conflicts
+	_, err = service.CreateCustomer(ctx, customer.CreateCustomerInput{
+		Namespace: s.namespace,
+		Customer: customer.Customer{
+			Name: TestName,
+			UsageAttribution: customer.CustomerUsageAttribution{
+				SubjectKeys: TestSubjectKeys,
+			},
+		},
+	})
+
+	require.ErrorAs(
+		t,
+		err,
+		&customer.SubjectKeyConflictError{Namespace: s.namespace, SubjectKeys: TestSubjectKeys},
+		"Creating a customer with same subject keys must return conflict error",
+	)
+}
+
+// TestUpdate tests the updating of a customer
+func (s *CustomerHandlerTestSuite) TestUpdate(ctx context.Context, t *testing.T) {
```

Comment:
This is not invoked from customer_test.go

test suits help with auto discovery as per here:
https://github.com/openmeterio/openmeter/pull/1541/files#diff-7a2c8eb899f11a37561acff33295dc9abe674dd76d818a3ff27348190c88a2c0